src/NHibernate/AdoNet/GenericBatchingBatcher.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System;
+using System.Data.Common;
+using System.Text;
+using NHibernate.AdoNet.Util;
+using NHibernate.Dialect;
+using NHibernate.Exceptions;
+using NHibernate.SqlCommand;
+
+namespace NHibernate.AdoNet
+{
+	/// <summary>
+	/// A generic batcher that will batch UPDATE/INSERT/DELETE commands by concatenating them with a semicolon.
+	/// Use this batcher only if there are no dedicated batchers in the given environment. Unfortunately some
+	/// database clients do not support concatenating commands with a semicolon. Here are the known clients
+	/// that do not work with this batcher:
+	/// - FirebirdSql.Data.FirebirdClient
+	/// - Oracle.ManagedDataAccess
+	/// - System.Data.SqlServerCe
+	/// </summary>
+	public partial class GenericBatchingBatcher : AbstractBatcher
+	{
+		private readonly int _maxNumberOfParameters = int.MaxValue;
+		private readonly BatchingCommandSet _currentBatch;
+		private int _totalExpectedRowsAffected;
+		private StringBuilder _currentBatchCommandsLog;
+
+		public GenericBatchingBatcher(ConnectionManager connectionManager, IInterceptor interceptor, char statementTerminator)
+			: base(connectionManager, interceptor)
+		{
+			BatchSize = Factory.Settings.AdoBatchSize;
+			StatementTerminator = statementTerminator;
+			_currentBatch = new BatchingCommandSet(this);
+			// On Sql Server there is a limit of 2100 parameters, but 2 are reserved for sp_executesql, so
+			// we are able to use up to 2098 parameters. When sp_prepexec is used for preparing and executing
+			// statements then one more parameter is used. Set the max number of parameters to 2097 in order
+			// to ensure that we will never exceed the limit.
+			if (Factory.Dialect is MsSql2000Dialect)
```

Comment:
Could this be added to Dialect itself? Eg `public virtual int? MaxNumberOfParameters {get;}`?

Reply:
Added.